src/varname.jl

```diff
@@ -104,78 +105,83 @@ julia> getsym(@varname(y))
+function Base.get(obj, vn::VarName{sym}) where {sym}
+    return getoptic(vn)(obj)
 end
 
 """
     set(obj, vn::VarName{sym}, value)
 
-Alias for `set(obj, PropertyLens{sym}() ∘ getlens(vn), value)`.
+Alias for `set(obj, PropertyLens{sym}() ∘ getoptic(vn), value)`.
 """
-function Setfield.set(obj, vn::VarName{sym}, value) where {sym}
-    return Setfield.set(obj, PropertyLens{sym}() ∘ getlens(vn), value)
+function Accessors.set(obj, vn::VarName{sym}, value) where {sym}
+    return Accessors.set(obj, PropertyLens{sym}() ∘ getoptic(vn), value)
 end
 
 
-Base.hash(vn::VarName, h::UInt) = hash((getsym(vn), getlens(vn)), h)
+Base.hash(vn::VarName, h::UInt) = hash((getsym(vn), getoptic(vn)), h)
 function Base.:(==)(x::VarName, y::VarName)
-    return getsym(x) == getsym(y) && getlens(x) == getlens(y)
+    return getsym(x) == getsym(y) && getoptic(x) == getoptic(y)
 end
 
 # Allow compositions with lenses.
-function Base.:∘(vn::VarName{sym,<:Lens}, lens::Lens) where {sym}
-    return VarName{sym}(getlens(vn) ∘ lens)
+function Base.:∘(vn::VarName{sym,T}, optic) where {sym,T}
+    return VarName{sym}(getoptic(vn) ⨟ optic)
+end
+function Base.:∘(vn::VarName{sym,typeof(identity)}, optic) where {sym}
+    return VarName{sym}(optic)
 end
 
-function Base.show(io::IO, vn::VarName{<:Any,<:Lens})
-    # No need to check `Setfield.has_atlens_support` since
-    # `VarName` does not allow dynamic lenses.
+function Base.show(io::IO, vn::VarName{sym,T}) where {sym,T}
     print(io, getsym(vn))
-    _print_application(io, getlens(vn))
+    _print_application(io, getoptic(vn))
 end
 
 # This is all just to allow to convert `Colon()` into `:`.
-_print_application(io::IO, l::Lens) = Setfield.print_application(io, l)
-function _print_application(io::IO, l::ComposedLens)
-    _print_application(io, l.outer)
-    _print_application(io, l.inner)
+# TODO: modify from https://github.com/JuliaObjects/Accessors.jl/blob/01528a81fdf17c07436e1f3d99119d3f635e4c26/src/sugar.jl#L502
+function _print_application(io::IO, optic)
+    opts = Accessors.deopcompose(optic)
+    inner = Iterators.takewhile(x -> applicable(Accessors._shortstring, "", x), opts)
+    outer = Iterators.dropwhile(x -> applicable(Accessors._shortstring, "", x), opts)
+    if !isempty(outer)
+        show(io, opcompose(outer...))
+        print(io, " ∘ ")
+    end
+    shortstr = reduce(Accessors._shortstring, inner; init="")
+    print(io, shortstr)
 end
-_print_application(io::IO, l::IndexLens) =
-    print(io, "[", join(map(prettify_index, l.indices), ","), "]")
-# This is a bit weird but whatever. We're almost always going to
-# `concretize` anyways.
-_print_application(io::IO, l::DynamicIndexLens) = print(io, l, "(_)")
+# TODO: type piracy https://github.com/JuliaObjects/Accessors.jl/blob/01528a81fdf17c07436e1f3d99119d3f635e4c26/src/sugar.jl#L495
+Accessors._shortstring(prev, o::IndexLens) = "$prev[$(join(map(prettify_index, o.indices), ", "))]"
+Accessors._shortstring(prev, o::typeof(identity)) = "$prev"
```

Comment:
Why do we need this? Maybe add a comment as to why we're overloading this stuff.

And do what we did before: if you're going to overload behavior of something we don't own, then define a new `_shortstring` method which defaults to `Accessors._shortstring` but can be further overloaded without piracy.

That is,
```julia
_shortstring(prev, o) = Accessors._shortstring(prev, o)
# Specializations
...
```
and then call this in `_print_application` instead.
and then